### PR TITLE
feat: add custom axis label style formatter

### PR DESCRIPTION
**Pull request by tomriddle25**
## PR Checklist:
- [x] I have tested this extensively and it does not break any existing behavior.
- [x] I have added/updated examples and tests for any new behavior.
- [x] If this is a significant change, an issue has already been created where the problem / solution was discussed: [N/A, or add link to issue here]
       

## PR Description

This PR adds a custom axis label style formatter for `BaseAxis`. The interface `IAxisStyleFormatter` can be used to define per index value styling ( textColor, typeFace, etc ) for the axis labels.

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -3,5 +3,10 @@ include 'MPChartLib'
 include 'MPChartExample'
 //include ':MPChartLib-Realm'
 //project(':MPChartLib-Realm').projectDir = new File('../MPAndroidChart-Realm/MPChartLib-Realm')
+include ':MPChartLib'
+include ':MPChartExample'
+project(':MPAndroidChart/MPChartLib').projectDir = new File(rootDir, 'MPAndroidChart/MPChartLib/')
+project(':MPAndroidChart/MPChartExample').projectDir = new File(rootDir, 'MPAndroidChart/MPChartExample/')
+
```
Comment (amanv8060) [inline settings.gradle]: we don't need this.

File: MPChartLib/src/main/java/com/github/mikephil/charting/renderer/XAxisRenderer.java [hunk 2]
```diff
@@ -230,6 +249,10 @@ protected void drawLabels(Canvas c, float pos, MPPointF anchor) {
     protected void drawLabel(Canvas c, String formattedLabel, float x, float y, MPPointF anchor, float angleDegrees) {
         Utils.drawXAxisValue(c, formattedLabel, x, y, mAxisLabelPaint, anchor, angleDegrees);
     }
+
+    protected void drawLabelWithFormatting(Canvas c, String formattedLabel, float x, float y, MPPointF anchor, float angleDegrees) {
+        Utils.drawXAxisValue(c, formattedLabel, x, y, mAxisLabelPaint, anchor, angleDegrees);
+    }
```
Comment (amanv8060) [inline MPChartLib/src/main/java/com/github/mikephil/charting/renderer/XAxisRenderer.java]: where is this function used ?

Comment (tomriddle25) [reply MPChartLib/src/main/java/com/github/mikephil/charting/renderer/XAxisRenderer.java]: Ah, need to remove. its not needed.

File: settings.gradle (hunk 1 above)
Comment (tomriddle25) [reply settings.gradle]: Okay.

**Review by amanv8060 — APPROVED** [review]

**Pull request merged**